src/docfx/lib/RetryUtility.cs

```diff
@@ -22,8 +22,7 @@ public static async Task<T> Retry<T>(Func<Task<T>> action, IEnumerable<Type> exc
                 {
                     return await action();
                 }
-                catch (Exception ex)
-                when (exceptions == null || exceptions.Any(e => e.IsInstanceOfType(ex)))
+                catch (Exception ex) when (isExpectedException == null || isExpectedException(ex))
```

Comment:
nit: `isExpectedException` should always be non null, so there is no need to do the null check here.